[FEAT] #217 작성하기 화면 코드 최신화 및 리팩토링

inddoni opened the pull request:
## 📌 관련 이슈
<!-- 관련있는 이슈 번호(#000)을 적어주세요.
  해당 pull request merge와 함께 이슈를 닫으려면 
  closed: #Issue_number를 적어주세요 -->
  closed: #217 

## 📌 변경 사항 및 이유
<!-- 변경한 내용과 그 이유를 적어주세요. -->
- 작성하기 화면(TVC 포함) 코드를 최신화했어요.
  - 접근제어자 전면 점검
  - Then 적용
  - 마크업 개선
  - 코드 가독성 개선 (self 붙이기, 코드 위치 변경) 등등

## 📌 PR Point
<!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->
- 기능은 변경되거나 추가된 부분은 없고 단순 리팩토링 작업이에요. (빅 PR이지만 내용 변화가 없어서 통으로 작업했어요🙏)
- 새로운 기능을 추가하기 앞서 코드 정리 및 로직 리팩토링을 하려고 했는데, 생각보다 양이 많아서 스타일적인 부분 위주로 정리했어요
  - 좀 더 디테일한 로직정리는 더 필요할 것 같아요

## 📌 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
- 앞으로 할 일
  - 안정성 향상을 위한 로직 개선
  - 이미지 피커 만들기
  - 이미지 피커 적용하기
  - 지은이가 만들어준 테마 하프뷰 붙이고 통신

inddoni committed:
♻️[REFACTOR]: #217 - CreatePostCourseTVC 코드 최신화

inddoni committed:
♻️[REFACTOR]: #217 - CreatePostDriveCourseTVC 코드 최신화

inddoni committed:
♻️[REFACTOR]: #217 - CreatePostParkingWarningTVC 코드 최신화

inddoni committed:
♻️[REFACTOR]: #217 - CreatePostPhotoTVC 코드 최신화

inddoni committed:
♻️[REFACTOR]: #217 - CreatePostThemeTVC 코드 최신화

inddoni committed:
♻️[REFACTOR]: #217 - CreatePostTitleTVC 코드 최신화

inddoni committed:
♻️[REFACTOR]: #217 - CreatePostVC 코드 최신화 및 리팩토링

hryeong66 commented on ChaRo-iOS/ChaRo-iOS/Source/Views/Cells/TVC/CreatePostScene/WrittingSence/CreatePostCourseTVC.swift:
```diff
@@ -40,54 +40,51 @@ class CreatePostCourseTVC: UITableViewCell {
-    }()
+    private let regionButton = UIButton().then {
+        $0.tag = 1
+        $0.setImage(UIImage(named: "unselect"), for: .normal)
```
p1;
저희 저번에 ImageLiterals로 이미지 정리해서 가져오기로 했어서 
요 부분 살짝 적용하면 좋을 것 같아요😊

inddoni replied:
https://github.com/TeamChaRo/ChaRo-iOS/pull/218/commits/c42c8d181872b892b3aaf9cd2ed9f81226970a76 여기서 리터럴 싹 적용했어요~

hryeong66 commented on ChaRo-iOS/ChaRo-iOS/Source/Views/Cells/TVC/CreatePostScene/WrittingSence/CreatePostCourseTVC.swift:
```diff
@@ -40,54 +40,51 @@ class CreatePostCourseTVC: UITableViewCell {
+        self.initTextField()
+        self.initPickerView()
+        self.setTextFieldAction()
     }
```
p2;
이 TVC는 xib 파일을 사용하지 않는것 같아서 xib삭제하고 init으로 변경하면 좋을 것 같아요😊

inddoni replied:
30f4ea55ee69b0c6d03ad5b84bc9183ff917d777 요기서 init으로 변경했어요~ 이것도 나중에 개선할 수 있으면 좋겠네유

hryeong66 commented on ChaRo-iOS/ChaRo-iOS/Source/Views/Cells/TVC/CreatePostScene/WrittingSence/CreatePostCourseTVC.swift:
```diff
@@ -96,114 +93,107 @@ class CreatePostCourseTVC: UITableViewCell {
+            self.city = self.filterList[currentIndex]
+            self.cityField.text = self.filterList[currentIndex]
+            self.cityField.textColor = .mainBlue
+            self.cityButton.setImage(UIImage(named: "select"), for: .normal)
```
p1;
여기도 수정 부탁드려요 :)

inddoni replied:
https://github.com/TeamChaRo/ChaRo-iOS/commit/c42c8d181872b892b3aaf9cd2ed9f81226970a76 여기서 리터럴 싹 적용했어요~

hryeong66 commented on ChaRo-iOS/ChaRo-iOS/Source/Views/Cells/TVC/CreatePostScene/WrittingSence/CreatePostCourseTVC.swift:
```diff
@@ -96,114 +93,107 @@ class CreatePostCourseTVC: UITableViewCell {
-            currentList = filterData.cityDict[filterList[0]]!
+            self.currentList = self.filterData.state
+        } else if index == 1 && self.filterList[0] != "" {
+            self.currentList = self.filterData.cityDict[filterList[0]]!
```
p3;
여기 강제 언래핑 되어있어서 ??로 사용하면 좋지 않을까요??😄

inddoni replied:
30f4ea55ee69b0c6d03ad5b84bc9183ff917d777 0번 인덱스에 직접 접근하는 것도 개선하는게 좋을 것 같긴한데 일단 강제 언래핑만 변경했어요~ 감사해요!

hryeong66 commented on ChaRo-iOS/ChaRo-iOS/Source/Views/Cells/TVC/CreatePostScene/WrittingSence/CreatePostCourseTVC.swift:
```diff
@@ -213,76 +203,89 @@ extension CreatePostCourseTVC {
-        regionField.snp.makeConstraints{
+        self.regionField.snp.makeConstraints {
             $0.height.equalTo(21)
             $0.leading.equalTo(regionButton.snp.leading).offset(12)
```
p4;
정말 사소한건데 height랑 witdh 설정 부분은 top, leading, trailing, bottom 설정하고 난 후 다음에 쓰는걸로 맞추는 건 어떨까요??😄

inddoni replied:
https://github.com/TeamChaRo/ChaRo-iOS/pull/218/commits/0d2fb70ed0f1069d7b0c429f682056a95b2809fb 요기서 높이, 너비가 아래오도록 순서 변경했어요~

hryeong66 commented on ChaRo-iOS/ChaRo-iOS/Source/Views/Cells/TVC/CreatePostScene/WrittingSence/CreatePostTitleTVC.swift:
```diff
@@ -102,50 +104,51 @@ extension CreatePostTitleTVC: UITextViewDelegate {
-        titleTextView.snp.makeConstraints{
+        self.titleTextView.snp.makeConstraints{
             $0.top.equalTo(self.snp.top).offset(23)
             $0.left.equalTo(self.snp.left).offset(20)
             $0.right.equalTo(self.snp.right).inset(20)
```
p2;
left right 보다는 leading, trailing이 더 좋지 않을까요??
그리고 self.snp.top 여렇게 하는 이유도 궁금해요 :)

inddoni replied:
https://github.com/TeamChaRo/ChaRo-iOS/pull/218/commits/0d2fb70ed0f1069d7b0c429f682056a95b2809fb 요기서 leading, trailing으로 변경했어요~

top은 타이틀 텍스트 뷰가 항상 맨 위라고 생각해서 top으로 적용해도 된다고 생각했어요!

inddoni committed:
ImageLiterals 적용

inddoni committed:
height, width 순서 변경 등 레이아웃 코드리뷰 반영

inddoni committed:
짜친 코드 수정

inddoni committed:
🔨[FIX]: #217 - 누락된 register 수정

##### 제목은 최대 50 글자까지만 입력 ############## -> |

# 본문은 위에 작성
######## 본문은 한 줄에 최대 72 글자까지만 입력 ########################### -> |

# 꼬릿말은 아래에 작성: ex) #이슈 번호

# --- COMMIT END ---
#- 🚑️[HOTFIX] : issue나, QA에서 급한 버그 수정에 사용
#- 🔨[FIX] : 버그, 오류 해결
#- ➕[ADD] : Feat 이외의 부수적인 코드 추가, 라이브러리 추가, 새로운 파일 생성 시
#- ✨[FEAT] : 새로운 기능 구현
#- ⚰️[DEL] : 쓸모없는 코드 삭제
#- 📝[DOCS] : README나 WIKI 등의 문서 개정
#- 💄[MOD] : storyboard 파일,UI 수정한 경우
#- ✅[CHORE] : 코드 수정, 내부 파일 수정
#- ✏️ [CORRECT] : 주로 문법의 오류나 타입의 변경, 이름 변경 등에 사용합니다.
#- 🚚[MOVE] : 프로젝트 내 파일이나 코드의 이동
#- ⏪️[RENAME] : 파일 이름 변경이 있을 때 사용합니다.
#- ⚡️[IMPROVE] : 향상이 있을 때 사용합니다.
#- ♻️[REFACTOR] : 전면 수정이 있을 때 사용합니다
#- 🔀[MERGE]: 다른브렌치를 merge 할 때 사용합니다.
# ------------------
#     제목 첫 글자를 대문자로
#     제목은 명령문으로
#     제목 끝에 마침표(.) 금지
#     제목과 본문을 한 줄 띄워 분리하기
#     본문은 "어떻게" 보다 "무엇을", "왜"를 설명한다.
#     본문에 여러줄의 메시지를 작성할 땐 "-"로 구분
# ------------------

hryeong66 reviewed (approved):
고생하셨습니다👍💕
